corrections/LengthScale_Corr.py

```diff
@@ -38,29 +59,37 @@ def doCorr(self,inData,configFile):
         print "Scaling coordinates with lengthscale factors"
 
         self.GetCorr(configFile)
-        
+
         self.PrintCorr()
 
     # apply correction here to coordinate, then write back into entry, check if this really changes value in calling function
 
         for entry in inData:
-            coord = entry.displacement
-            if 'X' in entry.scanName:
-                coord_corr = [a*self.LS_ScaleX for a in coord]
-            if 'Y' in entry.scanName:
-                coord_corr = [a*self.LS_ScaleY for a in coord]
+            correction = 1.0
+            for sbs in self.SingleBeamScans:
+                if sbs[0] in entry.scanName:
+                    if 'X' in sbs[0] and sbs[1] == 1:
+                        correction = self.LS_ScaleX1
+                    elif 'Y' in sbs[0] and sbs[1] == 1:
+                        correction = self.LS_ScaleY1
+                    elif 'X' in sbs[0] and sbs[1] == 2:
+                        correction = self.LS_ScaleX2
+                    elif 'Y' in sbs[0] and sbs[1] == 2:
+                        correction = self.LS_ScaleY2
+                    else:
+                        continue
+                    break
+            else:
```

Comment:
Yes, it does. In the for-loop, we test all the entries of SingleBeamScans whether they describe the current scan, and break the loop once we find one, such that the else-statement is not called. If we don't find an entry of SingleBeamScans that matches the current scan, the for-loop terminates through exhaustion of the list and thus calles the else-statement afterwards. So, for every scan, either the correction is set to one of the per-beam corrections in the loop, or to the appropriate average correction in the else-statement.

To be sure about this behaviour, I tested with this code that indeed a correction is only assigned once for all possible input scan names:

```python
def test(SingleBeamScans, scanName):
  correction = 1.0
  for sbs in SingleBeamScans:
    if sbs[0] in scanName:
      if 'X' in sbs[0] and sbs[1] == 1:
        print 'X1'
      elif 'Y' in sbs[0] and sbs[1] == 1:
        print 'Y1'
      elif 'X' in sbs[0] and sbs[1] == 2:
        print 'X2'
      elif 'Y' in sbs[0] and sbs[1] == 2:
        print 'Y2'
      else:
        continue
      break
  else:
    if 'X' in scanName:
      print 'avX'
    elif 'Y' in scanName:
      print 'avY'
SingleBeamScans = [["X3", 2], ["Y3", 2], ["X4", 1], ["Y4", 1]]
test(SingleBeamScans, 'X1')
test(SingleBeamScans, 'X3')
test(SingleBeamScans, 'Y2')
test(SingleBeamScans, 'Y4')
```